### ISSUE TITLE
A certain feed is always empty

### ISSUE THREAD

**Issue by kriegaex** [spenibus/rss-filter-php#7]
Can you tell me why, no matter whether I filter anything or not, basically passing through the original feed https://www.solcom.de/rss/projectfeed.xml, it always shows up empty in RSS Filter, despite the original feed having many entries? I would appreciate your help.

**Comment by spenibus** [comment]
Looks like CloudFlare protection. We're not deemed human enough.

````
HTTP/1.1 403 Forbidden
Date: Tue, 23 Apr 2024 11:11:23 GMT
Content-Type: text/html; charset=UTF-8
Transfer-Encoding: chunked
Connection: keep-alive
X-Frame-Options: SAMEORIGIN
Referrer-Policy: same-origin
Cache-Control: max-age=15
Expires: Tue, 23 Apr 2024 11:11:38 GMT
````

>Why have I been blocked?
>This website is using a security service to protect itself from online attacks. The action you just performed triggered the security solution. There are several actions that could trigger this block including submitting a certain word or phrase, a SQL command or malformed data.

Adding a browser agent to the config gets me a little further but not to the xml.

````
<userAgent>Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0</userAgent>
````

>Thank you very much for your interest.
>Unfortunately, this service is currently not available in your country.
>Thank you for your understanding.

Tried the script with both IPV4 and IPV6 but same result as above.

````
[url] => https://www.solcom.de/rss/projectfeed.xml
[primary_ip] => 2606:4700:20::681a:ca9
[primary_port] => 443

[headers] => HTTP/1.1 403 Forbidden
             Date: Tue, 23 Apr 2024 11:27:39 GMT
````

````
[url] => https://www.solcom.de/rss/projectfeed.xml
[primary_ip] => 172.67.74.12
[primary_port] => 443

[headers] => HTTP/1.1 403 Forbidden
             Date: Tue, 23 Apr 2024 11:27:04 GMT
````

Yet opening the url in firefox from the same machine does display the feed so I'm not really geo-blocked.

````
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
  <channel>
    <ttl>30</ttl>
    <atom:link href="https://www.solcom.de/rss/projectfeed.xml" rel="self" type="application/rss+xml" />
    <title><![CDATA[SOLCOM Projektangebote]]></title>
    <description><![CDATA[Aktuelle SOLCOM Projektangebote]]></description>
    <link><![CDATA[https://www.solcom.de/rss/projectfeed.xml]]></link>
    <copyright><![CDATA[SOLCOM GmbH]]></copyright>
````

Interestingly, tried the url in Firefox with the curl user agent and got the geo-block message.
The script does indeed use curl but since we got a block despite changing the user agent, the server must be expecting something more.

````
User-Agent: curl/8.6.0
````

In conclusion, the server expects something from the client to assess its legitimacy (ie. human) but I have no idea what.

**Comment by kriegaex** [comment]
Thanks for your analysis. I had a user agent string set from the beginning, but to no avail. I know that Solcom geo-blocks logins to the regular website, even in regular browsers. I noticed that when I was travelling overseas. But actually, now I am in Germany, and so is my web hosting provider - unless of course, the IP ranges are from elsewhere. I did not try this locally. I also noticed, just like you, that I can access the feed from Google Chrome, and my RSS feed collector service InoReader can also access the original feed without any problems despite also being an automated server-side solution.

**Comment by kriegaex** [comment]
I also just tried locally with

```text
 curl -A "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36" https://www.solcom.de/rss/projectfeed.xml
```

and it worked, my local IP obviously not being blocked. Where are you located?

One more thing: Can I make the script on my web hosting server write a log or display diagnostic information?

**Comment by spenibus** [comment]
> Where are you located?

France.

> Can I make the script on my web hosting server write a log or display diagnostic information?

There's nothing built-in, currently the only way is to dump vars with `print_r()` directly in the code.

Considering I can access the feed via a regular browser, it might just be some missing config in the curl request. I'll make some more tests if I find the time.
